x/treasury/abci.go

```diff
@@ -17,13 +17,17 @@ func EndBlocker(ctx sdk.Context, k Keeper) {
 	}
 
 	// Update luna issuance after finish all works
-	defer k.RecordHistoricalIssuance(ctx)
+	defer k.RecordEpochInitialIssuance(ctx)
```

Comment:
Why do we need to record the initial issuance of the epoch? Could we not just read the previous epoch issuance? (i.e. issuance at the last block of the previous epoch = issuance at the first block of the current epoch)

Reply:
That function is recording the issuance of the last block at epoch so after a epoch this recorded issuance is the previous epoch issuance. If we do not store the previous epoch issuance like this, there is no other way to read it.

Reply:
Understood.